docs/developer_guide.md

```diff
@@ -0,0 +1,75 @@
+# Developer guide
+
+## Leapp tool
+### Installing on RHEL 7
+
+**Register & subscribe**
+```
+subscription-manager register
+subscription-manager list --available --match-installed
+subscription-manager attach --pool POOL_ID
+```
+
+**Update to the latest `7.x` version**
+```
+yum clean all
+yum update
+reboot
+
+cat /etc/os-release
+# => RHEL is on latest version
+```
+
+**Install Leapp tool**
+```
+subscription-manager repos --enable rhel-7-server-rpms
+subscription-manager repos --enable rhel-7-server-extras-rpms
+
+yum install -y leapp
+```
+### Installing on CentOS 7
+
+!!! There is an error when running `leapp preupgrade` command on `CentOS 7`, better try it on `RHEL 7` !!!
+
+**Update to the latest `7.x` version**
+```
+yum clean all
+yum update
+reboot
+
+cat /etc/os-release
+```
+
+**Install Leapp**
+```
+yum-config-manager --add-repo https://copr.fedorainfracloud.org/coprs/g/oamg/leapp/repo/epel-7/group_oamg-leapp-epel-7.repo
+
+yum install yum-utils leapp
+```
+
+## Leapp commands
+
+**Pre-upgrade check**
+```
+leapp preupgrade
+
+# => /var/log/leapp/leapp-report.json (This one is parsed by foreman-leapp)
+# => /var/log/leapp/leapp-report.txt
+# => /var/log/leapp/leapp-preupgrade.log
+
+leapp preupgrade --verbose  # Enable verbose logging
+leapp preupgrade --debug    # Enable debug mode
+```
+For the report sample see the file in `report_samples` folder.
```

Comment:
should we copy the `report_samples/leapp-report.json` to `report_samples/leapp-report.json`

Reply:
Sorry I don't follow, both paths are same.

Reply:
oops, copy pasting is hard, I meant:
should the developer copy the `report_samples/leapp-report.json` to
`/var/log/leapp/leapp-report.json` in order to use this mocked report?